KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,367 @@
+            i += 1
+
+    a = 0
+    while a < len(khyper_header_chap):
+        if khyper_header_chap[a] == "memes":
+            del khyper_header_chap[a]
+            del k_hyper_sub_chap[a]
+        else:
+            a += 1
+
+    global sorterIIIcheck
+    chap9 = 0
+
+    if sorterIIIcheck[sortloc][0] == 0:
+        tempref = ref9_3
+        sorterIIIcheck[sortloc][0] += 1
+        chap9 = 1
+    else:
+        tempref = ref14_3
+        offset = sorterIIIcheck[sortloc][1]
+        if sep1 == 1:
+            offset = 8
+
+    for d in range(0, len(tempref)):  # check every section and subsection line
+        i = tempref[d]
+        line = str(chap[i])
+        if "\\section{" in line or "\\subsection{" in line:
+            if "\\subsection{" in line:
+                temp = line[12:line.find("}", 7)]
+            else:
+                temp = line[9:line.find("}", 7)]
+
+        if sep1 == 0:
+            if name_chap in line.lower():
+                hyper_subs_chap.append([tempref[d + 1]])  # appends the index for the line before following subsection
+                hyper_headers_chap.append(temp)  # appends the name of the section the hypergeo subsection is in
+
+                if temp in khyper_header_chap:
+                    try:
+                        chap[tempref[d + 1] - 1] += "\paragraph{\\bf KLS Addendum: " + word + "}"
+                        chap[tempref[d + 1] - 1] += k_hyper_sub_chap[k_hyp_index_iii + offset]
+                        if chap9 == 1:
+                            sorterIIIcheck[sortloc][1] += 1
+                        k_hyp_index_iii += 1
+                    except IndexError:
+                        print("Warning! Code has found an error involving section finding for '" + name_chap + "'.")
+        elif sep1 == 1:
+            if name_chap in line.lower() and "orthogonality relation" not in line:
+                hyper_subs_chap.append([tempref[d + 1]])  # appends the index for the line before following subsection
+                hyper_headers_chap.append(temp)  # appends the name of the section the hypergeo subsection is in
+
+                if temp in khyper_header_chap:
+                    try:
+                        chap[tempref[d + 1] - 1] += "\paragraph{\\bf KLS Addendum: " + word + "}"
+                        chap[tempref[d + 1] - 1] += k_hyper_sub_chap[k_hyp_index_iii + offset]
+                        k_hyp_index_iii += 1
+                    except IndexError:
+                        print("Warning! Code has found an error involving section finding for '" + name_chap + "'.")
+        elif sep1 == 2:
+            if "hypergeometric representation" in line.lower():
+                hyper_subs_chap.append([tempref[d + 1]])  # appends the index for the line before following subsection
+                hyper_headers_chap.append(
+                    temp)  # appends the name of the section the hypergeo subsection is in so we can compare
+
+                if temp in khyper_header_chap:
```

Comment:
Why do you have so many `temp`s? I can understand the occasional use of it as a variable, but `temp` appears much too frequently throughout the code.